### PR TITLE
Add About window

### LINKED ISSUE #439: View version of GRIP from within the application
We should add a help menu that also displays the version of grip that you are running.
Perhaps with other menu items that link to the issues and the wiki?

### PR DESCRIPTION
**Pull request by AustinShalit**
Adds an about window that uses FXML.

untitled

Closes #439

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
That looks awesome! Do you want to add a link to the github site somewhere?

**Comment by AustinShalit** [comment]
I added a button to bring the user to the github page.  The only possible issue is that I had to make Main a singleton pattern.  Let me know what you think...

File: ui/src/main/java/edu/wpi/grip/ui/Main.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
 
 public class Main extends Application {
 
+    private static Main instance;
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/Main.java]: Can you not do this.... Hahahahah

File: ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package edu.wpi.grip.ui;
+
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.text.Text;
+
+public class AboutDialogController {
+
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java]:
```
@Inject
private Main main
```

**Comment by AustinShalit** [comment]
Currently broken.  I have little (read: no) experience with dependency injection.  @JLLeitschuh Do you know why it is not working as written?

**Comment by JLLeitschuh** [comment]
I'll take a look tomorrow

**Comment by JLLeitschuh** [comment]
Looks weird now.
Looks Weird now

**Comment by SamCarlberg** [comment]
Buttons shouldn't have ellipses in the text, remove them.
Maybe change the text of the screensteps button to 'Documentation'?

**Comment by AustinShalit** [comment]
New Style.  Default:
untitled
Mouse on the github button:
untitled 2

**Comment by SamCarlberg** [comment]
Some issues I've found (running Windows 8.1, not sure if it's present on OS X or Linux)

- Buttons don't look aligned
- Font smoothing on the subtext is weird, the text is pretty blocky
- Gray background isn't centered properly and crowds out the text

image

image

**Comment by AustinShalit** [comment]
I updated the style again.  This _should_ fix the weird rectangle issue.  Still not sure why the font is/was being weird.

**Comment by AustinShalit** [comment]
Mac:
mac
Windows (With font bug):
windows

**Comment by AustinShalit** [comment]
Updated to solve the failing tests.

File: ui/src/main/resources/edu/wpi/grip/ui/AboutDialog.fxml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            <Label style="-fx-font-size: 12;" text="GRIP is licensed under a 3 Clause BSD License" textFill="#818181" />
+            <Pane HBox.hgrow="ALWAYS" />
+         </children>
+         <padding>
+            <Insets bottom="5.0" left="5.0" right="5.0" top="5.0" />
+         </padding>
+      </HBox>
+   </children>
+   <stylesheets>
+       <URL value="@roboto/Roboto.css" />
```
Comment (AustinShalit) [inline ui/src/main/resources/edu/wpi/grip/ui/AboutDialog.fxml]: This line needs to be removed with #590.  I recommend we merge that pr first.

**Comment by AustinShalit** [comment]
@SamCarlberg Can you please check the latest to see if smoothing is correct?

File: ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @FXML
+    void mousePressedGithubButton(MouseEvent event) {
+        main.getHostServices().showDocument("https://github.com/WPIRoboticsProjects/GRIP");
+    }
+
+    @FXML
+    void initialize() {
+        assert versionNumberLabel != null : "fx:id=\"versionNumberText\" was not injected: check your FXML file 'AboutDialog.fxml'.";
```
Comment (JLLeitschuh) [inline ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java]: Not sure if this is necessary.

Comment (SamCarlberg) [inline ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java]: It's not, should remove it.

**Comment by SamCarlberg** [comment]
Looks good to me

File: ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java (hunk 4 above)
Comment (AustinShalit) [inline ui/src/main/java/edu/wpi/grip/ui/AboutDialogController.java]: Removed in 4547ce4ba161ff0b2492fc4d0c7c01132bdf344e

**Comment by AustinShalit** [comment]
This is ready to merge

**Pull request merged**